terragrunt/aws/api/lambda.tf

```diff
@@ -1,13 +1,13 @@
 module "url_shortener_lambda" {
-  source                 = "github.com/cds-snc/terraform-modules?ref=v5.0.2//lambda"
+  source                 = "github.com/cds-snc/terraform-modules?ref=v5.1.5//lambda"
   name                   = "${var.product_name}-api"
   billing_tag_value      = var.billing_code
   ecr_arn                = var.ecr_repository_arn
   enable_lambda_insights = true
   image_uri              = "${var.ecr_repository_url}:${var.ecr_tag}"
   memory                 = 3008
   timeout                = 300
-
+  dead_letter_queue_arn  = aws_sns_topic.lambda_dead_letter.arn
```

Comment:
One little thing for this - you'll need to update the API lambda function's IAM policy to allow it to `sns:Publish` to this SNS topic.